nomad/structs/structs.go

```diff
@@ -7088,20 +7059,6 @@ func (d *DesiredTransition) ShouldForceReschedule() bool {
 	return d.ForceReschedule != nil && *d.ForceReschedule
 }
 
-const (
-	AllocDesiredStatusRun   = "run"   // Allocation should run
-	AllocDesiredStatusStop  = "stop"  // Allocation should stop
-	AllocDesiredStatusEvict = "evict" // Allocation should stop, and was evicted
-)
-
-const (
-	AllocClientStatusPending  = "pending"
-	AllocClientStatusRunning  = "running"
-	AllocClientStatusComplete = "complete"
-	AllocClientStatusFailed   = "failed"
-	AllocClientStatusLost     = "lost"
-)
```

Comment:
I have a very slight preference for keep these definitions next to where they're (indirectly) used (the Allocation struct).

I know this means manually copying to the api package, but these haven't changed in 2 years so copying isn't a huge cost.

That being said it's a very minor preference and not a blocker either way.

Reply:
It's generated! https://github.com/hashicorp/nomad/pull/5213/files#diff-7f97ec598f57966b8c13390fc38706deR3

Reply:
They're generated in `api/` from the file they're moved to in `nomad/structs`. My slight preference is to copy and paste these particular definitions instead of using generation to keep them grouped with related definitions in `nomad/structs/structs.go`, but it's a minor preference.

Reply:
Seeing that these values so rare and would be considered backward incompatible change to API package, I opted to inline them without having generation code.